app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,46 @@
+<h1>Recipe details</h1>
+<h1><%= @recipe.name %></h1>
+<p>Preparation time: <%= @recipe.preparation_time %> min</p>
+<p>Cooking time: <%= @recipe.cooking_time %> min</p>
+<p>Steps: <%= @recipe.description %></p>
+<% if can? :update, @recipe %>
+<%= button_to(@recipe.public? ? "Make Private" : "Make Public", toggle_recipe_public_path(@recipe), method: :put) %>
+    <%= link_to 'Generate Shopping List', "/shopping_list" %>
+    <%= link_to 'Add Ingredients', "/recipes/#{@recipe.id}/recipe_foods/new" %>
+<% end %>
+<div>
+<table>
+  <thead>
+    <tr>
+      <th>Food</th>
+      <th>Quantity</th>
+      <th>Value</th>
+      <% if can? :destroy, @recipe %>
+        <th>Actions</th>
+      <% end %>
+    </tr>
+  </thead>
+  <tbody>
+    <% @foods.each do |food| %>
+      <tr>
+        <td><%= link_to food.name %></td>
+        <% recipe_food = food.recipe_foods.find_by(food_id: food.id) %>
+        <% if recipe_food %>
+          <td><%= food.quantity %></td>
+          <td>
+            $<%= food.price * recipe_food.quantity %>
+          </td>
+        <% else %>
+          <td colspan="2">No quantity available</td>
+        <% end %>
+        <% if can? :destroy, @recipe %>
+          <td>
+            <%= link_to 'Edit', edit_recipe_recipe_food_path(@recipe, recipe_food) %>
+            <%= button_to 'Remove', recipe_recipe_food_path(@recipe, recipe_food), method: :delete %>
```

Comment:
- [ ] As mentioned by the previous :point_up: reviewer please make sure your page is the same as the given wireframe but still your design is not the same as the given wireframe so kindly fix it as well :slightly_smiling_face: